contracts/airdrop/StakingAirdrop.sol

```diff
@@ -0,0 +1,238 @@
+        serviceRegistry = _serviceRegistry;
+
+        // Assign airdrop amounts
+        for (uint256 i = 0; i < _serviceIds.length; ++i) {
+            // Check for already assigned amount
+            if (mapServiceIdAirdropAmount[_serviceIds[i]] > 0) {
+                revert NonZeroValue();
+            }
+
+            // Check for zero amount
+            if (_amounts[i] == 0) {
+                revert ZeroValue();
+            }
+
+            // Add to total airdrop amount
+            airdropAmount += _amounts[i];
+
+            // Record amount and service Id
+            mapServiceIdAirdropAmount[_serviceIds[i]] = _amounts[i];
+            serviceIds.push(_serviceIds[i]);
+        }
+    }
+
+    /// @dev Claims airdrop.
+    /// @notice Any `msg.sender` is able to trigger claim for eligible service Id.
+    /// @param serviceId Service Id.
+    function claim(uint256 serviceId) external {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        // Get service multisig
+        (, address multisig, , , , , ) = IServiceRegistry(serviceRegistry).mapServices(serviceId);
+
+        // Check multisig address
+        if (multisig == address(0)) {
+            revert ZeroAddress();
+        }
+        
+        // Get airdrop amount
+        uint256 amount = mapServiceIdAirdropAmount[serviceId];
+        
+        // Check for zero value
+        if (amount == 0) {
+            revert ZeroValue();
+        }
+
+        // Get contract balance
+        uint256 balance = IToken(token).balanceOf(address(this));
+        // Check for amount overflow
+        if (amount > balance) {
+            revert Overflow(amount, balance);
+        }
+
+        // Zero airdrop amount
+        mapServiceIdAirdropAmount[serviceId] = 0;
+
+        // Transfer airdrop tokens
+        IToken(token).transfer(multisig, amount);
+
+        emit Claimed(msg.sender, serviceId, multisig, amount);
+
+        _locked = 1;
+    }
+
+    /// @dev Claims airdrop for all eligible service Ids.
+    /// @notice Any `msg.sender` is able to trigger claim for eligible service Ids.
+    function claimAll() external {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        uint256 numServiceIds = serviceIds.length;
+        uint256[] memory localServiceIds = new uint256[](numServiceIds);
+        uint256[] memory localAmounts = new uint256[](numServiceIds);
+        uint256 totalAmount;
+
+        // Traverse all service Ids for amounts computation
+        for (uint256 i = 0; i < numServiceIds; ++i) {
+            localServiceIds[i] = serviceIds[i];
+
+            // Get airdrop amount for a specific service Id
+            uint256 amount = mapServiceIdAirdropAmount[localServiceIds[i]];
+
+            // Check for zero value
+            if (amount == 0) {
+                continue;
+            }
```

Comment:
does the indexing of the array account for the skips? please add tests for this case

Reply:
Tests are already there